item.rb

```diff
@@ -0,0 +1,28 @@
+require 'securerandom'
+require 'date'
+
+class Item
+  attr_accessor :label, :genre, :author
+  attr_reader :id, :published_date, :archived
+
+  def initialize(published_date = nil, id: nil)
+    @id = id || SecureRandom.uuid
+    @label = nil
+    @genre = nil
+    @author = nil
+    @published_date = published_date
+    @archived = false
+    move_to_archive
+  end
+
+  def can_be_archived?
+    return false if @published_date.nil?
+
+    published_date = DateTime.parse(@published_date)
+    published_date < DateTime.now - 30
+  end
+
+  def move_to_archive
+    @archived = can_be_archived?
+  end
```

Comment:
- [ ] As required for the project to correctly math all associations (1-to-many relationships) between classes, kindly ensure that the `Item` class has one `Genre`, `Author`, and `Label`. If you need a refresher about associations read [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships). For example:
```ruby
class Item
    attr_reader :label, :genre, :author
   .
   .
   .
   # Same example to each one(i.e. label and author)
   def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
    end
    ...
end
```